plugins/cri/cri.go

```diff
@@ -64,20 +64,26 @@ func init() {
 				return nil
 			}
 			const pluginName = string(plugins.GRPCPlugin) + ".cri"
-			original, ok := pluginConfigs[pluginName]
+			src, ok := pluginConfigs[pluginName].(map[string]interface{})
 			if !ok {
 				return nil
 			}
-			src := original.(map[string]interface{})
 
-			// Currently only a single key migrated
```

Comment:
> // Currently only a single key migrated

@dmcgowan I wonder if this is intended or not. Appreciate any pointer if there are other changes required to migrate other keys.

Reply:
I'm trying to follow the history on this one, it was moved before the migration so it might have just been missed https://github.com/containerd/containerd/commit/65b3922df75de299fe95a53a74fed8835740d6ba

I'm wondering if this copy approach is the best, it seems just to be stripping out the other values but that shouldn't be necessary. Ideally when values are migrated the migration cleans up the value in the original struct, so if these values aren't moving, they shouldn't need to be copied.

Reply:
>  Ideally when values are migrated the migration cleans up the value in the original struct

That means a field can only be used by 1 plugin at a time. If a field (say `config.containerd.runtime`) is used by 2 _independent_ plugins (e.g., cri-runtime, cri-image) during migration, neither of them can clean up the value in the original struct (cri-grpc). 

Then when we map the original struct to the new cri-grpc `ServerConfig` struct, it will print some confusing warnings (unknown key in TOML) https://github.com/containerd/containerd/issues/10959.

Reply:
Yeah, you are right, the cri config is pretty messy and migration isn't that straight forward. Always overwriting like you are doing here is fine